test/helpers/helpers.go

```diff
@@ -334,3 +348,104 @@ func CheckCscSuccessfulDeletion(client test.FrameworkClient, cscName, namespace,
 
 	return nil
 }
+
+// RestartMarketplace scales the marketplace deployment down to zero and then scales
+// it back up to it's original number of replicas, and waits for a successful deployment.
+func RestartMarketplace(client test.FrameworkClient, namespace string) error {
+	marketplace := &apps.Deployment{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: "marketplace-operator", Namespace: namespace}, marketplace)
+	if err != nil {
+		return err
+	}
+	replicas := marketplace.Spec.Replicas
+	zero := int32(0)
+	marketplace.Spec.Replicas = &zero
+	err = client.Update(context.TODO(), marketplace)
+	if err != nil {
+		return err
+	}
+	err = client.Get(context.TODO(), types.NamespacedName{Name: "marketplace-operator", Namespace: namespace}, marketplace)
+	if err != nil {
+		return err
+	}
+	marketplace.Spec.Replicas = replicas
+	err = client.Update(context.TODO(), marketplace)
+	if err != nil {
+		return err
+	}
+	err = WaitForSuccessfulDeployment(client, *marketplace)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// CreateDatastoreCsc returns a newly built CatalogSourceConfig
+func CreateDatastoreCsc(name, namespace string) *v2.CatalogSourceConfig {
+	lables := make(map[string]string)
+	lables[datastore.DatastoreLabel] = "true"
+
+	return &v2.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: fmt.Sprintf("%s/%s",
+				v1.SchemeGroupVersion.Group, v1.SchemeGroupVersion.Version),
+			Kind: v2.CatalogSourceConfigKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels:    lables,
+		},
+		Spec: v2.CatalogSourceConfigSpec{
+			TargetNamespace: namespace,
+			Packages:        "",
+		},
+	}
+}
+
+// CreateInstalledCsc returns a newly built CatalogSourceConfig
+func CreateInstalledCsc(namespace string) *v2.CatalogSourceConfig {
+	name := TestInstalledCscPublisherName + namespace
+	return &v2.CatalogSourceConfig{
```

Comment:
Doesn't this require some labels too?

Reply:
What labels were you thinking of? I don't think we need the owner labels here because they are added to the child resources by the operator.

Reply:
I was thinking of any labels that the UI would add and that you are looking for in the upgrade code.

Reply:
I don't see any use of labels in the UI code. We just use the naming convention `installed-...`

Reply:
We only need these labels set on the subscription object so that the csc can be found. This is being done correctly below in the `CreateSubscription` function